hbase/src/main/scala/akka/stream/alpakka/hbase/internal/HBaseFlowStage.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.hbase.internal
+
+import akka.stream._
+import akka.stream.alpakka.hbase.HTableSettings
+import akka.stream.stage._
+import org.apache.hadoop.hbase.client.Table
+
+import scala.util.control.NonFatal
+import scala.util.{ Failure, Success, Try }
+
+private[hbase] class HBaseFlowStage[A](settings: HTableSettings[A])
+    extends GraphStage[FlowShape[A, A]]
+    with HBaseCapabilities {
+
+  override protected def initialAttributes: Attributes =
+    Attributes.name("HBaseFLow").and(ActorAttributes.dispatcher("akka.stream.default-blocking-io-dispatcher"))
+
+  private val in = Inlet[A]("messages")
+  private val out = Outlet[A]("result")
+
+  override val shape = FlowShape(in, out)
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+    new GraphStageLogic(shape) with StageLogging {
+
+      implicit val connection = connect(settings.conf)
+
+      val table: Table = getOrCreateTable(settings.tableName, settings.columnFamilies)
+
+      var completionState: Option[Try[Unit]] = None
+
+      setHandler(out, new OutHandler {
+        override def onPull() =
+          tryPull(in)
```

Comment:
As we do not do any detaching in this stage now, it can just do a pull here (we will never get two pulls in a row)